src/uniparc/components/entry/XRefsFacets.tsx

```diff
@@ -0,0 +1,169 @@
+          query: priorityBatch.map(mapper).sort().join(' OR '),
+          size: priorityBatch.length,
+        })
+      : undefined;
+
+    const extraBatch = facetValues.slice(
+      ITEMS_DISPLAYED,
+      MAX_API_SIZE + ITEMS_DISPLAYED
+    );
+    const extraURL = extraBatch.length
+      ? getAPIQueryUrl({
+          namespace: Namespace.taxonomy,
+          query: extraBatch.map(mapper).sort().join(' OR '),
+          size: extraBatch.length,
+        })
+      : undefined;
+    return [facetValues, priorityURL, extraURL];
+  }, [xrefs]);
+
+  const priorityData = useDataApi<{ results: Taxonomy[] }>(priorityURL);
+  // wait for the priority to have finished before loading the rest
+  const extraData = useDataApi<{ results: Taxonomy[] }>(
+    priorityData.loading ? undefined : extraURL
+  );
+
+  const taxonomyMap = useMemo(
+    () =>
+      new Map(
+        [
+          ...(priorityData.data?.results || []),
+          ...(extraData.data?.results || []),
+        ].map((taxon) => [`${taxon.taxonId}`, taxon])
+      ),
+    [priorityData.data, extraData.data]
+  );
+  return (
+    <Facets>
+      <Facet
+        data={useMemo(
+          () => ({
+            label: 'Status',
+            name: 'active',
+            values: Object.entries(countBy(xrefs, (xref) => xref.active))
+              .map(([value, count]) => ({
+                label: value === 'true' ? 'Active' : 'Inactive',
+                value,
+                count,
+              }))
+              .sort(sortByCount),
+          }),
+          [xrefs]
+        )}
+      />
+      <Facet
+        data={useMemo(
+          () => ({
+            label: 'Organisms',
+            name: 'organisms',
+            values: taxonomyFacetValues.map(({ value, count }) => {
+              const taxon = taxonomyMap.get(value);
+              return {
+                label: taxon ? (
+                  <span
+                    title={`${taxon.commonName ? `${taxon.commonName}, ` : ''}${
+                      taxon.scientificName
+                    }, taxon ID ${value}`}
+                  >
+                    {taxon.commonName || taxon.scientificName}
+                  </span>
+                ) : (
+                  value
+                ),
+                value,
+                count,
+              };
+            }),
+          }),
+          [taxonomyFacetValues, taxonomyMap]
+        )}
+      />
+      <Facet
+        data={useMemo(
+          () => ({
+            label: 'Databases',
+            name: 'databases',
+            values: Object.entries(countBy(xrefs, (xref) => xref.database))
+              .map(([value, count]) => {
+                let label: ReactNode = value;
+                const entryType = getEntryTypeFromString(value);
+                if (entryType === EntryType.REVIEWED) {
+                  label = (
+                    <>
+                      <EntryTypeIcon entryType={entryType} />
+                      UniProtKB reviewed
```

Comment:
This is also used in XrefSection (and maybe other parts of the website), maybe make it a string enum? Same with unreviewed.

Reply:
Well, there's actually a bug here, where `TREMBLNEW` is detected as `UniProtKB unreviewed` ('TREMBLNEW' is distinct from `TrEMBL`). I'll refactor the logic in the XRefsSections in order to be reused here.

Reply:
done